ts/features/fims/saga/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { SagaIterator } from "redux-saga";
+import { call, put, select, takeLatest } from "typed-redux-saga";
+import { ActionType } from "typesafe-actions";
+import { fimsTokenSelector } from "../../../store/reducers/authentication";
+import { fimsDomainSelector } from "../../../store/reducers/backendStatus";
+import {
+  mockHttpNativeCall,
+  mockSetNativeCookie
+} from "../../pushNotifications/sagas/mockFIMSCallbacks";
+import {
+  fimsGetConsentsListAction,
+  fimsGetRedirectUrlAndOpenBrowserAction
+} from "../store/actions";
+import { fimsCTAUrlSelector } from "../store/selectors";
+
+export function* watchFimsSaga(): SagaIterator {
+  yield* takeLatest(
+    fimsGetConsentsListAction.request,
+    handleFimsGetConsentsList
+  );
+  yield* takeLatest(
+    fimsGetRedirectUrlAndOpenBrowserAction.request,
+    handleFimsGetRedirectUrlAndOpenBrowser
+  );
+}
+
+function* handleFimsGetConsentsList() {
+  // action: ActionType<(typeof fimsGetConsentsListAction)["request"]>
+  // ) {
+  //   NavigationService.navigate(FIMS_ROUTES.MAIN, {
+  //     screen: FIMS_ROUTES.CONSENTS
+  //   }); // TODO:: maybe move navigation in saga from utils
+
+  const fimsToken = yield* select(fimsTokenSelector);
+  const oIDCProviderUrl = yield* select(fimsDomainSelector);
+  const fimsCTAUrl = yield* select(fimsCTAUrlSelector);
+
+  // in paper it looks cleaner to use FP-TS for this, but it does not play well with generator functions
+  if (
```

Comment:
during development I used to check like this, but to be honest I do not trust js' nullish check and am afraid that something like an empty string would make this statement true.

Also, I believe being more granular is not a bad call; what do you think?

Reply:
I read it this way: "if we do not have a valid fims token, or a valid provider url of a valid cta url, then report an error at the beginning of the flow" (meaning, it cannot even start).

When is a FIMS token (provider / CTA) not valid? When we have no data or the data content is not valid. Latter is verified by server but former we know that undefined, null, empty, ..., better said, nullish values are not valid. That is why a "!variable" covers all cases where we do not start the flow since we know for sure that something is amiss (and the error can be specialized).